src/librustc_incremental/persist/hash.rs

```diff
@@ -39,6 +39,14 @@ impl<'a, 'tcx> HashContext<'a, 'tcx> {
         }
     }
 
+    pub fn is_hashable(dep_node: &DepNode<DefId>) -> bool {
```

Comment:
I'd prefer if this was called `is_input_node` or `is_graph_root` to make it clearer what we are after.